rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactory.java

```diff
@@ -42,7 +54,35 @@ protected String getBarcodeValsAsCSV(Intent data) {
     }
 
     @Override
-    protected void sensorTriggered() {
-        ((RDTJsonFormFragmentPresenter) widgetArgs.getFormFragment().getPresenter()).moveToNextStep("step2");
+    protected void moveToNextStep(Date expDate) {
+
+        if (isSensorTriggered()) {
+            String triggeredPageAddr = stepStateConfig.getStepStateObj().optString(Constants.Step.SENSOR_TRIGGERED_PAGE, "step1");
+            JsonFormFragment nextFragment = RDTJsonFormFragment.getFormFragment(triggeredPageAddr);
+            widgetArgs.getFormFragment().transactThis(nextFragment);
+        } else {
+            super.moveToNextStep(expDate);
+        }
+    }
+
+    private boolean isSensorTriggered() {
+        boolean result = false;
+
+        try {
+            JsonApi jsonApi = (JsonApi) widgetArgs.getContext();
+            String stepName = widgetArgs.getStepName();
+            JSONObject stepObject = jsonApi.getStep(stepName);
+            JSONArray fields = stepObject.getJSONArray(JsonFormUtils.FIELDS);
+            for (int i = 0; i < fields.length(); i++) {
+                JSONObject fieldObject = fields.getJSONObject(i);
+                if (TEMP_SENSOR.equals(fieldObject.getString(JsonFormUtils.KEY))) {
+                    result = fieldObject.optBoolean(JsonFormUtils.VALUE, false);
+                }
+            }
```

Comment:
It should be possible to get the temp sensor information from the barcode https://github.com/onaio/rdt-standard/blob/3be90ebdfadd9191c4b7cdcf2c1ba65db4ad7347/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactory.java#L46.

Reply:
https://github.com/onaio/rdt-standard/blob/3be90ebdfadd9191c4b7cdcf2c1ba65db4ad7347/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java#L65

Reply:
You can add another method `moveToNextStep(Date expDate, boolean isSensorTriggered)`  in `CovidRDTBarcodeFactory`. This method will check whether sensor is triggered and also call the original `moveToNextStep(Date expDate)`.

Reply:
i have override `moveToNextStep` method and added a data argument.